server/src/main/java/org/opensearch/extensions/rest/RegisterRestActionsRequest.java

```diff
@@ -23,29 +23,29 @@
  * @opensearch.internal
  */
 public class RegisterRestActionsRequest extends TransportRequest {
-    private String nodeId;
+    private String uniqueId;
```

Comment:
I went through this thought process and that led me to my suggestion in [SDK #95](https://github.com/opensearch-project/opensearch-sdk-java/issues/95).

The central problem we have to solve is what to put in the URI string for the REST request: `curl -X GET localhost:9300/_extensions/_what_do_we_put_here/index`

Here are the possible identifiers we could use:
 - The extension name.  This is probably the easiest to add to the URI, but this is unlikely to be unique and in theory a user could have multiple same-named extensions.  Perhaps they use one AnomalyDetection extension for one type of data and a different AnomalyDetection extension for another.
 - The ephemeral id.  This is what we should be generating on-the-fly for each execution of OpenSearch, and will change from run to run.  But the user should not have to look-up and then encode this ID in their REST request. Would you expect a user to type `curl -X GET localhost:9200/_extensions/3477fd34-4c08-4b9f-80fd-83284b65b854/index`?
 - The host:port combination.  This would work and the user is already using one such combination (`localhost:9200`) but I doubt they'll be able to remember which port all their different extensions are running on.
 - The node identification.  Similar to ephemeral ID this is not something the user would know, and a long string of text/code to try to copy/paste.
 - The unique ID.  This is a user-configurable identification of the extension, as short or as long as they want to make it.  They could copy the extension name in their config if all their extensions are uniquely named, in which case it's the easy "name" version. If they do have same-named extensions they can deconflict them here.  In the example above with two Anomaly Detection extensions they could call them ad1 and ad2, and just type `curl -X GET localhost:9200/_extensions/ad1/index`.   I suspect when we eventually get to a point of publishing a catalog, an "automatic/default" unique name could be reverse-DNS based like the Maven/Gradle dependency (and plugin) ecosystem.

So for now I think Unique ID is the best.  If we need to change it later (before version 1.0) we can.  At this point I don't know what else it's going to be used for and this seems like the best one.  We could add a new restUriString field to serve the same purpose, if we really needed it.

Reply:
We can follow the same naming convention here for `_what_do_we_put_here` the way we have for the current Plugins API. Example for Notification plugin we have an API like: 
```
GET /_plugins/_notifications/features
```
Similarly, we can have
```
GET /_extensions/_anomaly_detection/create_detector
```

Reply:
> We can follow the same naming convention here for `_what_do_we_put_here` the way we have for the current Plugins API. 

That's essentially the same as using the name, which I discussed in [this comment](https://github.com/opensearch-project/OpenSearch/pull/4282#discussion_r955500116). In the plugin system we only have a small number and they all have unique names.  

The "REST name" needs to be unique in the event someone has two same-named extensions. I envision a lot of name collisions in a public name registry.  If we enforce unique names we risk "name squatting" or abandoned extensions with the good names, who just happened to get there early.

@saratvemulapalli notes in [Comment on SDK #60](https://github.com/opensearch-project/opensearch-sdk-java/pull/60#discussion_r930265844) that we may run the same extension on two nodes and that may be the purpose of the uniqueId, but I think that applies more to the ephemeralId.

[SDK #65](https://github.com/opensearch-project/opensearch-sdk-java/issues/65) discussing what to use uniqueId for (among other things) has been open for 4 weeks without resolution.  I'd like to not [bikeshed](https://en.wikipedia.org/wiki/Law_of_triviality) here and just use UniqueId for now. We can change it later.

Reply:
Sure @dbwiddis that absolutely makes sense from a customer experience. Thanks for putting this out there. 

Thanks for brining up https://github.com/opensearch-project/opensearch-sdk-java/issues/65. This conversation really helps and probably we can close it down.